docs/rules/jsx-no-script-url.md

```diff
@@ -0,0 +1,22 @@
+# Prevent usage of `javascript:` URLs (react/jsx-no-script-url)
+
+**In React 16.9** any URLs starting with `javascript:` [scheme](https://wiki.whatwg.org/wiki/URL_schemes#javascript:_URLs) log a warning. 
+React considers the pattern as a dangerous attack surface, see [details](https://reactjs.org/blog/2019/08/08/react-v16.9.0.html#deprecating-javascript-urls). 
+**In a future major release**, React will throw an error if it encounters a `javascript:` URL.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```jsx
+<a href="javascript:"></a>
+<a href="javascript:void(0)"></a>
+<a href="j\n\n\na\rv\tascript:"></a>
+```
+
+The following patterns are **not** considered warnings:
+
+```jsx
+<Foo href="javascript:"></Foo>
```

Comment:
what about a `Link` component? we should provide a way to indicate that a specific list of custom components is like an "a", and also to override the name of the "href" prop on those elements.

Reply:
@ljharb Exports can be renamed during import, so `Link` component can be imported as `Foo`:
```js
import { Link as Foo } from "react-router-dom";
```
How are you going to handle it?
Moreover if you pass `javascript:void(0)` as `to` property to  `Link`  component from `react-router-dom`, it doesn't set the same value as `href` attribute:
```jsx
<Link to="javascript:void(0)">Link</Link>

// it adds '/' and doesn't run js
<a href="/javascript:void(0)">Link</a>
```

Reply:
It'd be by name; if you rename it to `Link2` it wouldn't error, to `Link` it would.

That's certainly one example of a `Link` component, but not one I've ever seen since blindly prepending `/` would break some kinds of URLs.

Reply:
@ljharb The PR has been updated, take a look when you get a chance.

Reply:
@ljharb are there any updates here?

Reply:
i'll rereview, but i'm waiting to hear from @gaearon before merging this.

Reply:
@ljharb We have over two weeks since the last commit. Is there any chance to be merged?

Reply:
@ljharb we have over month since the last discussion. Are there any concerns to merge it? The rule is disabled by default and I don't see how it can affect anybody, but it can help to prevent React errors. I think @gaearon had enough time to take a look at this.

Reply:
@ljharb Any chance to merge?